### DISCUSSION TITLE
variable types are not strict on urql

### DISCUSSION THREAD

**Discussion by NickSurmanidze** [dotansimha/graphql-code-generator#9802] [Q&A]
We recently switched from typescrypt-urql to client preset and based on docs variables should be typed. But they seem to allow adding additional variables:

```
  const [{ data, fetching }] = useQuery({
    query: GetMyItemsDocument,
    variables: {
      pagination: {
        limit: 10,
        offset:  0,
        myExtraVariable: 'Oops!', // no lint error on extra variable
      },
    },
  });
```

As a workaround we add 
```
useQuery<
    ResultOf<typeof GetMyItemsDocument>,
    VariablesOf<typeof GetMyItemsDocument>
  >(...)
  ```
  and now we get lint error for extra variables. 
  
Did anyone else experience similar issue?

**Comment by maxcan** [comment]
Any luck here?  it seems like Urql defaults teh type of useQuery and useMutation to any so passing in a stricter type as an argument doesn't narrow anything
